crates/client/rpc/src/lib.rs

```diff
@@ -943,48 +944,88 @@ where
             }
         }
 
+        fn extract_fee_and_contract_address(
+            events: &Vec<starknet_core::types::Event>,
+            sequencer_address: &Felt252Wrapper,
+        ) -> (FieldElement, Option<FieldElement>) {
+            let transfer_event_key = starknet_core::utils::get_selector_from_name("Transfer").unwrap();
+            let find_transfer_fee_event = events.iter().find_map(|event| {
+                event.keys.iter().find(|&&key| key == transfer_event_key)?;
+                event.data.get(1).and_then(|send_to| (*send_to == sequencer_address.0).then_some(()))?;
+                Some(event)
+            });
+
+            match find_transfer_fee_event {
+                Some(transfer_fee_event) => {
+                    let actual_fee = Felt252Wrapper(transfer_fee_event.data[2].clone()).0; // ony low bits, see execute_fee_transfer() why
+                    let contract_address = Felt252Wrapper(transfer_fee_event.data[0].clone()).0;
+                    (actual_fee, Some(contract_address))
+                }
+                None => (Default::default(), None), // transaction without fee
+            }
+        }
+
+        let events_converted: Vec<starknet_core::types::Event> =
+            events.clone().into_iter().map(event_conversion).collect();
+
+        let (actual_fee, contract_address) = extract_fee_and_contract_address(&events_converted, &sequencer_address);
```

Comment:
With your current impl of `extract_fee_and_contract_address` I an account does a transfer to the sequencer address. You will register this tx as the fee transfer, when there is another fee transfer after.

Reply:
Maybe you should iterate in reverse. Or always assume transfer will be the last event of the tx and directly index it. Rather than searching for it

Reply:
Write a test for this scenario

Reply:
Thank you for this idea! 
About test. Seems that [this test](https://github.com/keep-starknet-strange/madara/blob/10f6179101926067de4fb114b4510d1737f8f7bb/crates/pallets/starknet/src/tests/events.rs#L57) is about correct order of events already.

Reply:
I meant to write tests about transferring ETH to the sequencer address and make sure that it doesn't mess up with get_transaction_receipt